contracts/NonfungiblePositionManagerV4.sol

```diff
@@ -120,7 +130,37 @@ contract NonfungiblePositionManagerV4 is
         checkDeadline(params.deadline)
         returns (uint256 tokenId, uint128 liquidity, uint256 amount0, uint256 amount1)
     {
-        // TODO: implement this
+        (liquidity, amount0, amount1) = addLiquidity(
+            AddLiquidityParams({
```

Comment:
this re-defining of AddLiquidityParams is kinda ugly... i think i'd prefer either:

1. We could have AddLiqParams as a sub-struct of MintParams like
```
struct MintParams {
   AddLiquidityParams addLiqParams
   address recipient
   uint256 deadline
}
```

orrrr
2. We could not even bother with the `MintParams` entirely and just have
```
function mint(AddLiquidityParams calldata params, address recipient, uint256 deadline)
```

Reply:
then these 12 lines could just be like `(liquidity, amount0, amount1) = addLiquidity(params);` 😏

Reply:
++ option 2